### PR TITLE
docs: update instruction for conandata format and patching

### PR DESCRIPTION
**Pull request by prince-chrismc**
Docs!

this bring the "How to" inline with some other recent doc changes

### CODE REVIEW

File: docs/how_to_add_packages.md [hunk 1]
```diff
@@ -113,34 +113,27 @@ versions:
...
-    sha256: "8c48baf3babe0d505d16cfc0cf272589c66d3624264098213db0fb00034728e9"
-  "1.1.1":
-    url: "https://www.url.org/source/mylib-1.0.1.tar.gz"
-    sha256: "15b6393c20030aab02c8e2fe0243cb1d1d18062f6c095d67bca91871dc7f324a"
-patches:
-  "1.1.1":
-    - patch_file: "patches/1.1.1-001-simpler-cmakelists.patch"
-      base_path: "source_subfolder"
-```
+A detailed breakdown of all the fields can be found in [conandata_yml_format.md](conandata_yml_format.md). We strongly encourage adding the [patch feilds](conandata_yml_format.md#patches-fields) to help track where patches come from and what issue they solve.
```
Comment (jgsogo) [inline docs/how_to_add_packages.md:118]: The file `conandata_yml_format.md` is not added in this PR

Comment (jgsogo) [inline docs/how_to_add_packages.md:118]:
```suggestion
A detailed breakdown of all the fields can be found in [conandata_yml_format.md](conandata_yml_format.md). We strongly encourage adding the [patch fields](conandata_yml_format.md#patches-fields) to help track where patches come from and what issue they solve.
```

File: docs/how_to_add_packages.md [hunk 2]
```diff
@@ -113,34 +113,27 @@ versions:
...
 
 Inside the `conanfile.py` recipe, this data is available in a `self.conan_data` attribute that can be used as follows:
 
 ```py
 def source(self):
-     tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+    tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
Comment (jgsogo) [inline docs/how_to_add_packages.md:124]: Use tools from v1/v2 compatible layer: `conan.tools.files.get(self,...)`

File: docs/how_to_add_packages.md [hunk 3]
```diff
@@ -113,34 +113,27 @@ versions:
...
+     
+def export_sources(self):
+    for patch in self.conan_data.get("patches", {}).get(self.version, []):
+        self.copy(patch["patch_file"])
```
Comment (jgsogo) [inline docs/how_to_add_packages.md:128]: Same here, there is standalone `copy` function

**Review by jgsogo — CHANGES_REQUESTED** [review]

File: docs/how_to_add_packages.md (hunk 1 above)
Comment (prince-chrismc) [reply docs/how_to_add_packages.md:118]: It's already merged https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md

it's from #11713 

just removing the stale old reference about it

File: docs/how_to_add_packages.md (hunk 2 above)
Comment (prince-chrismc) [reply docs/how_to_add_packages.md:124]: Can I do this in a separate PR? 🙏 

A TON of docs needs v2 migration attention and I dont want to mix more changes

Comment (prince-chrismc) [reply docs/how_to_add_packages.md:124]:
```suggestion
    files.get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
```

File: docs/how_to_add_packages.md (hunk 3 above)
Comment (prince-chrismc) [reply docs/how_to_add_packages.md:128]:
```suggestion
        files.copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
```

File: docs/how_to_add_packages.md [hunk 4]
```diff
@@ -113,34 +113,27 @@ versions:
...
 
 def build(self):
-    for patch in self.conan_data.get("patches", {}).get(self.version, []):
-        tools.patch(**patch)
+    apply_conandata_patches(self)
```
Comment (prince-chrismc) [inline docs/how_to_add_packages.md:131]:
```suggestion
    files.apply_conandata_patches(self)
```

File: docs/reviewing.md [hunk 5]
```diff
@@ -107,8 +107,7 @@ For simple cases, `tools.replace_in_file` is allowed.
 
 ```py
 def _patch_sources(self):
-    for patch in self.conan_data.get("patches", {}).get(self.version, []):
-        tools.patch(**patch)
+    apply_conandata_patches(self)
```
Comment (prince-chrismc) [inline docs/reviewing.md:110]:
```suggestion
    files.apply_conandata_patches(self)
```

**Review by prince-chrismc — COMMENTED** [review]
v2 changes

**Review by jgsogo — APPROVED** [review]
👍 , step by step

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**